bin/test_exercises.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# ###
+# test_exercises.sh, by Tim Austin (neenjaw) tim@neenjaw.com
+# ###
+# Script is to test each exercise against it's test unit individually
+# against the example solution to prove the test works, and the problem
+# is solvable.
+#
+# This works by creating a copy of the exercises, then for each exercise running
+# `mix test` with some options, suppressing the output, then tallying the results
+# with a brief report.
+#
+# An exit code of 0 indicates all tests successful, 1 indicates an error with at
+# least one exercise.
+# ###
+
+# helper subroutine
+test_or_tests () {
+    local tests="test"
+    if (( $1 != 1)); then
+        tests+="s"
+    fi
+    printf "%s" "$tests"
+}
+
+# Initialize counts / array
+test_count=0
+pass_count=0
+fail_count=0
+
+failing_exercises=()
+
+# Store the relative root dir to variable
+relative_root=$(pwd)
+
+# make a copy of exercises
+cp -a exercises tmp-exercises
+
+# test each exercise
+for exercise in tmp-exercises/*
+  do if [ -d $exercise ]; then
+    cd "$exercise"
+
+    exercise_name=$(basename $exercise)
+    test_count=$((test_count+1))
+
+    printf "Testing: $exercise_name"
+
+    # Move the example into the lib file
+    for file in lib/*.ex
+      do mv "$file" "${file/.ex/.ex.bkp}"
+    done
+
+    mv example.exs lib/example.ex
+
+    # do testing then based on exit code print the result
+    mix test --no-elixir-version-check --include pending > /dev/null
+
+    if [ "$?" -eq 0 ]; then
+        printf " -- pass ✅\n"
+        pass_count=$((pass_count+1))
+    else
+        printf " -- fail ❌\n"
```

Comment:
In the failing case, it would be nice to know what actually failed. The simplest thing would be to change
```
mix test --no-elixir-version-check --include pending > /dev/null
```
to
```
mix test --color --include pending > output
```
cat output
```
after this `printf`.

Reply:
True, would save you a step, I like it, will add it in.

Reply:
New version will output the following on fail:

```
➜ bin/test_exercises.sh
Testing: accumulate -- fail ❌
  * test accumulate upcased strings (5.7ms)
  * test accumulate square numbers (4.2ms)
  * test accumulate reversed strings (186.6ms)
  * test nested accumulate (0.00ms)
  * test accumulate empty list (0.00ms)
Testing: acronym -- pass ✅
Testing: allergies -- pass ✅
Testing: all-your-base -- pass ✅
Testing: alphametics -- pass ✅
Testing: anagram -- pass ✅
```

Reply:
This is added in now.  I think in the future it might be handy to have a custom result formatter for this purpose as the output (while readable to humans) isn't as easily grepped for the results and so it feels a bit janky matching on ANSI color codes to determine the individually failing tests, but it works for now.